features/src/autogluon/features/generators/text_ngram.py

```diff
@@ -194,13 +195,16 @@ def _generate_ngrams(self, X, downsample_ratio: int = None):
 
             if not self._is_fit:
                 transform_matrix = self._adjust_vectorizer_memory_usage(transform_matrix=transform_matrix, text_data=text_data, vectorizer_fit=vectorizer_fit, downsample_ratio=downsample_ratio)
+                nlp_features_names = vectorizer_fit.get_feature_names()
+                nlp_features_names_final = np.array([f'{nlp_feature}.{x}' for x in nlp_features_names]
+                                                    + [f'{nlp_feature}._total_']
+                                                    )
+                self._feature_names_dict[nlp_feature] = nlp_features_names_final
 
-            nlp_features_names = vectorizer_fit.get_feature_names()
-
-            X_nlp_features = pd.DataFrame(transform_matrix.toarray())  # TODO: Consider keeping sparse
-            X_nlp_features.columns = [f'{nlp_feature}.{x}' for x in nlp_features_names]
-            X_nlp_features[nlp_feature + '._total_'] = X_nlp_features.gt(0).sum(axis=1).astype(np.int32)
-
+            transform_array = transform_matrix.toarray()
+            nonzero_count = np.count_nonzero(transform_array, axis=1).astype(np.uint16)
```

Comment:
Would uint16 max value a problem on the large datasets, where counts can overflow above 65535?

Reply:
Theoretically yes, but it would need to have 66k unique tokens present in a single row, which doesn't seem very likely to ever happen.

If it did happen, then the user has bigger problems to deal with than this feature overflowing (like the fact that the individual ngram counts are uint8, so if a single ngram appears > 511 times, it will overflow the counter).

Reply:
Added inline comment